go/vt/discovery/tablet_stats_cache.go

```diff
@@ -19,132 +19,44 @@ package discovery
 import (
 	"sync"
 
-	"golang.org/x/net/context"
-	"vitess.io/vitess/go/vt/log"
 	querypb "vitess.io/vitess/go/vt/proto/query"
+
+	"vitess.io/vitess/go/vt/log"
 	topodatapb "vitess.io/vitess/go/vt/proto/topodata"
-	"vitess.io/vitess/go/vt/srvtopo"
-	"vitess.io/vitess/go/vt/topo"
 	"vitess.io/vitess/go/vt/topo/topoproto"
 )
 
-// TabletStatsCache is a HealthCheckStatsListener that keeps both the
-// current list of available TabletStats, and a serving list:
+// tabletStatsCache is an internal data structure that keeps both the
+// current list of available tabletStats, and a serving list:
 // - for master tablets, only the current master is kept.
-// - for non-master tablets, we filter the list using FilterByReplicationLag.
+// - for non-master tablets, we filter the list using FilterStatsByReplicationLag.
 // It keeps entries for all tablets in the cell(s) it's configured to serve for,
 // and for the master independently of which cell it's in.
 // Note the healthy tablet computation is done when we receive a tablet
 // update only, not at serving time.
-// Also note the cache may not have the last entry received by the tablet.
+// Also note the cache may not have the last entry received from the tablet.
 // For instance, if a tablet was healthy, and is still healthy, we do not
 // keep its new update.
-type TabletStatsCache struct {
-	// cell is the cell we are keeping all tablets for.
-	// Note we keep track of all master tablets in all cells.
-	cell string
-	// ts is the topo server in use.
-	ts *topo.Server
+type tabletStatsCache struct {
 	// mu protects the following fields. It does not protect individual
 	// entries in the entries map.
-	mu sync.RWMutex
+	mu sync.Mutex
 	// entries maps from keyspace/shard/tabletType to our cache.
 	entries map[string]map[string]map[topodatapb.TabletType]*tabletStatsCacheEntry
```

Comment:
I had thought about key'ing it off querypb.Target, but that includes cell. We have 1 cache per cell so that they can be updated in parallel rather than 1 cache across all cells.

Reply:
This doesn't apply any more since we have deleted tabletStatsCache and replaced with a map-map of *TabletHealth.